bcda/main_test.go

```diff
@@ -237,6 +237,57 @@ func (s *MainTestSuite) TestCreateToken() {
 	buf.Reset()
 }
 
+func (s *MainTestSuite) TestGenerateClientCredentialsCLI() {
```

Comment:
Is there a benefit to having multiple test cases in one function? For me, it's clearer when there's one case per function. I know we haven't followed a consistent pattern for how we group tests though. Maybe a topic for the team to discuss.

Reply:
I think its generally bad practice.  Ideally, we have one-to-one mapping of test case to test function.  I'd like to not re-arrange all of our tests as part of this PR; maybe a discussion for watercooler?